### PR TITLE
[net6] Bump DotNetPreview version to 6.0.100-preview.1.21109.8

### PR DESCRIPTION
**Pull request by radekdoulik**
Bump `DotNetPreview` version to `6.0.100-preview.1.21109.8`

Add linker reflection based workaround, till
the [new API](https://github.com/mono/linker/pull/1774) is ready to use.
It forces the `ILLink` to process all the referenced assemblies in our steps,
without it, no assemblies are processed before `MarkStep`.

Updated apk size reference files. The NET6 linker has more optimizations
enabled, assembly sizes decreased about 0.5% and native shared libraries
are smaller as well - about 0.7%.

Comparison of `BuildReleaseArm64` apk test sizes before/after
```
Simple XA NET6

Summary:
  +           0 Other entries 0.00% (of 52,514)
  +           0 Dalvik executables 0.00% (of 316,876)
  -       3,237 Assemblies -0.47% (of 690,451)
  -      34,992 Shared libraries -0.67% (of 5,202,312)
  -       7,680 Uncompressed assemblies -0.57% (of 1,357,824)
  -      10,670 Package size difference -0.36% (of 2,927,336)

XForms XA NET6

Summary:
  +           0 Other entries 0.00% (of 750,947)
  +           0 Dalvik executables 0.00% (of 3,454,884)
  -      29,117 Assemblies -0.51% (of 5,696,363)
  -      34,992 Shared libraries -0.66% (of 5,277,504)
  -      66,048 Uncompressed assemblies -0.51% (of 12,839,424)
  -      36,556 Package size difference -0.37% (of 9,899,330)

Simple XA legacy monodroid

Summary:
  +           0 Other entries 0.00% (of 52,485)
  +           0 Dalvik executables 0.00% (of 316,848)
  +           1 Assemblies 0.00% (of 1,153,531)
  +           0 Shared libraries 0.00% (of 10,510,880)
  +           0 Uncompressed assemblies 0.00% (of 2,765,312)
  -         116 Package size difference -0.00% (of 5,001,423)

XForms XA legacy monodroid

Summary:
  +           0 Other entries 0.00% (of 750,918)
  +           0 Dalvik executables 0.00% (of 3,454,264)
  -           2 Assemblies -0.00% (of 4,674,321)
  +           0 Shared libraries 0.00% (of 10,582,264)
  +           0 Uncompressed assemblies 0.00% (of 11,293,696)
  -          12 Package size difference -0.00% (of 10,476,653)
```

### CODE REVIEW

**Comment by radekdoulik** [comment]
Testing on CI, looks like illink lost some steps.

File: src/Microsoft.Android.Sdk.ILLink/SetupStep.cs [hunk 1]
```diff
@@ -43,9 +46,13 @@ protected override void Process ()
...
-			InsertAfter (new FixAbstractMethodsStep (cache), "RemoveUnreachableBlocksStep");
-			InsertAfter (subSteps2, "RemoveUnreachableBlocksStep");
-			InsertAfter (subSteps1, "RemoveUnreachableBlocksStep");
+			InsertAfter (new FixAbstractMethodsStep (cache), "SetupStep");
+			InsertAfter (subSteps2, "SetupStep");
+			InsertAfter (subSteps1, "SetupStep");
+
+			// temporary workaround: this call forces illink to process all the assemblies
+			foreach (var assembly in (IEnumerable<AssemblyDefinition>)getReferencedAssembliesMethod.Invoke (Context, null))
+				Context.LogMessage ($"Reference assembly to process: {assembly}");
```
Comment (dellis1972) [inline src/Microsoft.Android.Sdk.ILLink/SetupStep.cs:55]: Is this something we can make use of https://github.com/xamarin/xamarin-android/pull/5317/files#diff-be3e7a80e4a796e2a4af1bbb778767f3f7d9c431534156e771e4ff7dc885e5c5R12 to do? (once its merged)

File: src/Microsoft.Android.Sdk.ILLink/SetupStep.cs [hunk 2]
```diff
@@ -43,9 +46,13 @@ protected override void Process ()
...
 
-			InsertAfter (new FixAbstractMethodsStep (cache), "RemoveUnreachableBlocksStep");
-			InsertAfter (subSteps2, "RemoveUnreachableBlocksStep");
-			InsertAfter (subSteps1, "RemoveUnreachableBlocksStep");
+			InsertAfter (new FixAbstractMethodsStep (cache), "SetupStep");
+			InsertAfter (subSteps2, "SetupStep");
+			InsertAfter (subSteps1, "SetupStep");
+
+			// temporary workaround: this call forces illink to process all the assemblies
+			foreach (var assembly in (IEnumerable<AssemblyDefinition>)getReferencedAssembliesMethod.Invoke (Context, null))
```
Comment (jonathanpeppers) [inline src/Microsoft.Android.Sdk.ILLink/SetupStep.cs:54]: Should we have a `null` check for  `getReferencedAssembliesMethod` and throw, just so if the method disappears we'll get a better error?

File: src/Microsoft.Android.Sdk.ILLink/SetupStep.cs (hunk 1 above)
Comment (radekdoulik) [reply src/Microsoft.Android.Sdk.ILLink/SetupStep.cs:55]: yes, we can use this call to replace `GetAssembliesStep`. it wouldn't live long though, the workaround will be most likely around just for few weeks till we have new API available.

File: src/Microsoft.Android.Sdk.ILLink/SetupStep.cs (hunk 2 above)
Comment (radekdoulik) [reply src/Microsoft.Android.Sdk.ILLink/SetupStep.cs:57]: Should not harm, even if it is short lived temporary.

**Comment by radekdoulik** [comment]
I would like to finish it with `6.0.100-preview.1.21109.8` so that I don't have to remeasure it and fix possible new issues. Will try the preview2 in new PR after that one. Hopefully it will be less bumpy than this one.

**Review by dellis1972 — APPROVED** [review]

**Pull request merged**